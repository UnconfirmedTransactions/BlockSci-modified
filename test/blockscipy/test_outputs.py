import pytest


def test_p2pkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-2-tx"])
    output_regression(tx, regtest)


def test_p2sh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-0-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-1-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    output_regression(tx, regtest)


def output_regression(tx, regtest):
    for out in tx.outputs:
        print(out.address, file=regtest)
        print(out.address_type, file=regtest)
        print(out.block, file=regtest)
        print(out.index, file=regtest)
        print(out.is_spent, file=regtest)
        print(out.spending_input, file=regtest)
        print(out.spending_tx, file=regtest)
        print(out.spending_tx_index, file=regtest)
        print(out.tx, file=regtest)
        print(out.tx_index, file=regtest)
        print(out.value, file=regtest)

def test_output_input_mapping(chain):
    for block in chain:
        for tx in block:
            for output in tx.outputs:
                if output.is_spent:
                    assert output == output.spending_input.spent_output
                    assert tx == output.spending_input.spent_tx
