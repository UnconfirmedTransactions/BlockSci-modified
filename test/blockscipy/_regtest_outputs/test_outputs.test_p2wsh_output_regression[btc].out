WitnessScriptHashAddress(bcrt1qektr66tljls59h4w5qwshr6fwr4qfeuwm3zkwcsd424sd046n92sagvgwa, wrapped_address=PubkeyAddress(mus5uwj8dVT9Wkvipg5YbLjhBa1uAs16Ui))
Pay to witness script hash
Block(tx_count=3, height=121, header_hash=6b4c87d26c48371ebb9a5561e226bb722c9263fc6c9c9f072ef67a8c3d0e5104, version=536870912, timestamp=1535767200, bits=545259519, nonce=0)
0
True
TxIn(spent_tx_index=144, address=WitnessScriptHashAddress(bcrt1qektr66tljls59h4w5qwshr6fwr4qfeuwm3zkwcsd424sd046n92sagvgwa), value=100000000)
Tx(len(txins)=1, len(txouts)=1, size_bytes=113, block_height=121, tx_index=145)
145
Tx(len(txins)=1, len(txouts)=2, size_bytes=235, block_height=121, tx_index=144)
144
100000000
PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)
Pay to pubkey hash
Block(tx_count=3, height=121, header_hash=6b4c87d26c48371ebb9a5561e226bb722c9263fc6c9c9f072ef67a8c3d0e5104, version=536870912, timestamp=1535767200, bits=545259519, nonce=0)
1
False
None
None
None
Tx(len(txins)=1, len(txouts)=2, size_bytes=235, block_height=121, tx_index=144)
144
4899990000
